org.eclipse.buildship.ui.test/src/main/groovy/org/eclipse/buildship/ui/view/execution/ExecutionViewDoubleClickTest.groovy

```diff
@@ -0,0 +1,47 @@
+package org.eclipse.buildship.ui.view.execution
+
+import org.eclipse.buildship.ui.test.fixtures.SwtBotSpecification
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem
+
+class ExecutionViewDoubleClickTest extends BaseExecutionViewTest {
+
+    def "Double-clicking group nodes expand and collapse children"() {
+        setup:
+        def project = sampleProject()
+        importAndWait(project)
+        launchTask('${workspace_loc:/root}', 'foo')
+        waitForConsoleOutput()
+
+        tree = getCurrentTree()
+
+        SWTBotTreeItem root = tree.getTreeItem('Run build')
+
+        expect:
+        root.expanded
+
+        when:
+        root.doubleClick()
+
+        then:
+        !root.expanded
+
+        when:
+        root.doubleClick()
+
+        then:
+        root.expanded
+
+        when:
+        SWTBotTreeItem runNode =  root.getNode('Run tasks')
+
+        then:
+        !runNode.expanded
+
+        when:
+        runNode.doubleClick()
+
+        then:
+        runNode.expanded
+    }
+
```

Comment:
I think we should have an additional test here verifying that double-clicking a test class and method opens the test file. Something like:
```
setup: importProjectWithTest()
when: testNode.doubleClick()
then: assertEditorOpenedOnFile(testFile)
cleanup: closeAllEditors()
```

Reply:
done